tracer/tracer_test.go

```
@@ -0,0 +1,67 @@
+	assert.IsType(t, &segment{}, seg, "Default segment should be of type *segment")
+
+	assert.NotPanics(t, func() { seg.Close(nil) })
+	assert.Nil(t, seg.AddMetadata("key", "value"))
+	assert.Nil(t, seg.AddError(errors.New("test error")))
+}
+
+func TestTracer_BeginSubSegment_WithFn(t *testing.T) {
+	mockSeg := new(MockSegment)
+	tracer := New()
+	ctx := context.Background()
+	expectedCtx := context.WithValue(ctx, "key", "value")
```

golangci-lint (1.64.8)

SA1029: should not use built-in type string as key for value; define your own type to avoid collisions

(staticcheck)